### PR TITLE
Use a decimal type for Transaction inflow, outflow, and amountMinor

### PR DESCRIPTION
**Pull request by jasonpetsod**
Install the decimal.js-light library to provide the decimal type. Convert all
call sites to pass in Decimal to Transaction().

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,6 +9,7 @@
     "babel-preset-es2015": "^6.22.0",
     "babel-preset-react": "^6.22.0",
     "babelify": "^7.3.0",
+    "decimal.js-light": "^2.2.0",
```
Comment (witriew) [inline package.json:12]: Is there a reason to not enforce precision to 2 (and other sorts of configs for Decimal)?

File: src/Transaction.jsx [hunk 2]
```diff
@@ -1,36 +1,43 @@
...
-      this.amountMinor = inflow * 100;
-    } else if (outflow > 0) {
-      // TODO: Stop using floats for money.
-      this.amountMinor = -1 * outflow * 100;
+    if (inflow.isZero() && outflow.isZero()) {
+      this.amountMinor = new Decimal('0');
+    } else if (inflow.isPositive()) {
+      this.amountMinor = inflow.times(100);
+    } else if (outflow.isPositive()) {
+      this.amountMinor = outflow.times(-100);
```
Comment (witriew) [inline src/Transaction.jsx]: Use `.neg()`; http://mikemcl.github.io/decimal.js-light/#neg.

File: src/Transaction.jsx [hunk 3]
```diff
@@ -1,36 +1,43 @@
...
-    if (inflow === 0 && outflow === 0) {
-      this.amountMinor = 0;
-    } else if (inflow > 0) {
-      // TODO: Stop using floats for money.
-      this.amountMinor = inflow * 100;
-    } else if (outflow > 0) {
-      // TODO: Stop using floats for money.
-      this.amountMinor = -1 * outflow * 100;
+    if (inflow.isZero() && outflow.isZero()) {
+      this.amountMinor = new Decimal('0');
```
Comment (witriew) [inline src/Transaction.jsx]: Initialize this at the top and you can drop this test.

Comment (witriew) [inline src/Transaction.jsx]: Is there a reason to be doing the `int <-> Decimal` type conversion here?  I think this goes better when it gets written to/read from Firebase.

File: package.json (hunk 1 above)
Comment (jasonpetsod) [reply package.json:12]: I don't think we need to force a precision during intermediate computation. When we output to the browser we round to two decimal places anyway.

File: src/Transaction.jsx (hunk 3 above)
Comment (jasonpetsod) [reply src/Transaction.jsx]: Done.

Comment (jasonpetsod) [reply src/Transaction.jsx]: Moved the conversion into MutableTransactionRow in 5119bbd1b62f4beaf3df5045c9174eb1a8255116.

File: src/Transaction.jsx (hunk 2 above)
Comment (jasonpetsod) [reply src/Transaction.jsx]: Done.

File: package.json (hunk 1 above)
Comment (witriew) [reply package.json:12]: Will we ever have a use-case where we need more precision during intermediate computation (I don't see us doing multiplication and division, only adds/subs)?  If not, I'd rather not have to ever worry about storing ridiculous floats.

**Review by witriew — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (jasonpetsod) [reply package.json:12]: Filed issue #35.

**Pull request merged**